op-e2e/e2eutils/challenger/helper.go

```diff
@@ -141,17 +144,17 @@ func applyCannonConfig(c *config.Config, t *testing.T, rollupCfg *rollup.Config,
 	c.Cannon.RollupConfigPath = rollupFile
 }
 
-func WithCannon(t *testing.T, rollupCfg *rollup.Config, l2Genesis *core.Genesis) Option {
+func WithCannon(t *testing.T, rollupCfg *rollup.Config, l2Genesis *core.Genesis, allocType e2econfig.AllocType) Option {
```

Comment:
It's unfortunate we need to pass the AllocType in here. Given it's available from the system along with rollup.Conifg and core.Genesis I think we can avoid it and just pass in System (but will need to use an interface - it's a pattern we've used elsewhere).  I can follow up after this is merged. Would love to avoid needing to call `AllocTypeFromEnv()` as much as possible.